pypeclub/OpenPype
# Photoshop: Move implementation to OpenPype

## Brief description
Move Photoshop integration to OpenPype codebase.

## Description
Photoshop was moved to OpenPype codebase and modified meanwhile a little bit.

## Changes
- moved photoshop implementation to OpenPype
- added `MainThreadItem` which stores callback and args + kwargs for function call
- removed `__init__.py` from empty hooks folder
- removed `__init__.py` and `lib.py` from plugins dir
    - moved content of `lib.py` to `api/plugin.py`
- getting of stub in loaders does not happen in global scope instead of that default `PhotoshopLoader` was implement which has `get_stub` method to retrieve that
- changed import of `main` in launching script
- few formatting changes

## Notes & TODOs (Next PR/s)
- didn't modify `README.md` which seems to be outdated anyway
- extension does contain avalon icon which should be probably changed to OpenPype?
- `Creator` class which was in `~/avalon/photoshop/pipeline.py` is not used at all so should probably be removed?
- I didn't test it yet - any fixes can be commited to this branch

## Testing notes:
- [ ] Normal usage:
1. Run photoshop integration
2. Check if works

- [ ] Webpublisher(Farm?) usage:
1. Run photoshop integration
2. Check if works

Related PR https://github.com/pypeclub/avalon-core/pull/410
